packages/react-form/src/hooks/list/dynamiclist.ts

```diff
@@ -0,0 +1,180 @@
+import {useMemo, useEffect, ChangeEvent, useState} from 'react';
+import isEqual from 'fast-deep-equal';
+
+import {
+  ValidationDictionary,
+  Validator,
+  NormalizedValidationDictionary,
+  FieldStates,
+  FieldDictionary,
+  FieldState,
+  ErrorValue,
+  ListValidationContext,
+} from '../../types';
+import {mapObject, normalizeValidation, isChangeEvent} from '../../utilities';
+
+import {
+  updateAction,
+  updateErrorAction,
+  newDefaultAction,
+  reinitializeAction,
+  resetAction,
+  useListReducer,
+  addFieldsAction,
+  removeFieldsAction,
+} from './reducer';
+
+interface DynamicList<Item extends object> {
+  fields: FieldDictionary<Item>[];
+  addField(): void;
+  removeField(index: number): void;
+}
+
+export function useDynamicList<Item extends object>(
+  initialList: Item[],
```

Comment:
Would it be possible to keep the same signature as `useList`, where you can either pass in the list of fields of a config object? 
`listOrConfig: FieldListConfig<Item> | Item[],`

What was the reason to separate them?

Reply:
Ahh yes that's fair! I think I separated them for simplicity when testing but will put them back together